### PR TITLE
feat(CVSB-11815): Specialist vehicle view

### PR DESCRIPTION
**Pull request by m-ujaffer**
recreated PR for re-approvals.

### CODE REVIEW

**Comment by JonLazariniDD** [comment]
comments on PR made here

https://github.com/dvsa/cvs-app-vtm/pull/174

File: src/app/technical-record/vehicle-summary/vehicle-summary-psv/vehicle-summary-psv.component.spec.ts [hunk 1]
```diff
@@ -12,17 +12,22 @@ describe('PsvVehicleSummaryComponent', () => {
...
       approvalType: 'approval',
       variantNumber: '123',
       ntaNumber: '4566',
+      speedLimiterMrk: true,
+      tachoExemptMrk: true,
+      euroStandard: '6',
+      fuelPropulsionSystem: 'gas',
+      numberOfWheelsDriven: 4,
+      emissionsLimit: 23,
       vehicleClass: TESTING_UTILS.mockBodyType()
```
Comment (alleypa) [inline src/app/technical-record/vehicle-summary/vehicle-summary-psv/vehicle-summary-psv.component.spec.ts:31]: use this instead `vehicleClass: {
        code: '2',
        description: 'motorbikes over 200cc'
      }`

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.spec.ts [hunk 2]
```diff
@@ -39,15 +39,13 @@ describe('VehicleSummaryComponent', () => {
...
-    expect(component.formatVehicleClassDescription()).toEqual('The first letter should be capital');
+    expect(component.formatVehicleClassDescription()).toEqual(
+      'The first letter should be capital'
+    );
   });
 
   it('should create format the vehicle class description', () => {
     fixture.detectChanges();
     expect(component.vehicleClassDescription).toEqual('The first letter should be capital');
   });
```
Comment (alleypa) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.spec.ts:50]: for this test please use this `  it('should create format the vehicle class description', () => {
    fixture.detectChanges();
    expect(component.vehicleClassDescription).toEqual('Motorbikes over 200cc');
  });`

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.spec.ts [hunk 3]
```diff
@@ -39,15 +39,13 @@ describe('VehicleSummaryComponent', () => {
...
 
   it('should check if axles has no parking brake mrk', () => {
-    expect(component.formatVehicleClassDescription()).toEqual('The first letter should be capital');
+    expect(component.formatVehicleClassDescription()).toEqual(
+      'The first letter should be capital'
+    );
   });
```
Comment (alleypa) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.spec.ts:45]: delete this test. not needed

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.ts [hunk 4]
```diff
@@ -9,7 +10,7 @@ import { TechRecord } from '@app/models/tech-record.model';
 export class VehicleSummaryComponent implements OnInit {
   @Input() activeRecord: TechRecord;
   vehicleClassDescription = '-';
-  constructor() {}
+  constructor(public techRecHelpers: TechRecordHelpersService) {}
```
Comment (alleypa) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.ts:13]: you need to mock this service in you test and return your desired value

File: src/app/technical-record/tech-record-helpers.service.spec.ts [hunk 5]
```diff
@@ -64,8 +61,21 @@ describe('TechRecordHelpersService', () => {
...
+    it('should return true if vehicle type is a standard one(trl)', () => {
+      expect(service.isStandardVehicle('trl')).toBe(true);
+    });
+
+    it('should return false if vehicle type is a motorcycle', () => {
+      expect(service.isStandardVehicle('motorcycle')).toBe(false);
+    });
+  });
+
   afterAll(() => {
```
Comment (alleypa) [inline src/app/technical-record/tech-record-helpers.service.spec.ts:78]: we can clean this up

File: src/app/technical-record/technical-record.component.html [hunk 6]
```diff
@@ -413,7 +421,11 @@ <h1 class="govuk-heading-xl title">Technical record</h1>
         <div
           class="auto-pad"
           [@.disabled]="true"
-          *ngIf="activeRecord.vehicleType === 'trl' || activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
+          *ngIf="
+            activeRecord.vehicleType === 'trl' ||
+            activeRecord.vehicleType === 'hgv' ||
+            activeRecord.vehicleType === 'psv'
+          "
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.html:428]: do want to change this to `techRecHelpers.isStandardVehicle(activeRecord.vehicleType`

File: src/app/technical-record/notes/notes.component.html [hunk 7]
```diff
@@ -1,22 +1,31 @@
...
 
-  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
+  <div
+    class="govuk-summary-list__row"
+    *ngIf="activeRecord.vehicleType !== 'trl' && activeRecord.vehicleType !== 'hgv'"
+  >
     <dt class="govuk-summary-list__key">Dispensations</dt>
     <dd id="test-dispensations" class="govuk-summary-list__value">
-      <span id="test-notes" class="notes-section">{{ activeRecord.dispensations | DefaultNullOrEmpty }}</span>
+      <span id="test-notes" class="notes-section">{{
```
Comment (m-ujaffer) [inline src/app/technical-record/notes/notes.component.html:26]: remove id="test-notes". should be fixed after rebase with common.

File: src/app/technical-record/notes/notes.component.html [hunk 8]
```diff
@@ -1,22 +1,31 @@
...
+        activeRecord.notes | DefaultNullOrEmpty
+      }}</span>
     </dd>
   </div>
 
   <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
     <dt class="govuk-summary-list__key">Remarks</dt>
     <dd id="test-dispensations" class="govuk-summary-list__value">
-      <span id="test-notes" class="notes-section">{{ activeRecord.remarks | DefaultNullOrEmpty }}</span>
+      <span id="test-notes" class="notes-section">{{
```
Comment (m-ujaffer) [inline src/app/technical-record/notes/notes.component.html:14]: remove id="test-notes". should be fixed after rebase with common.

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.ts [hunk 9]
```diff
@@ -9,7 +10,7 @@ import { TechRecord } from '@app/models/tech-record.model';
...
 
   ngOnInit() {
     this.vehicleClassDescription =
```
Comment (m-ujaffer) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.ts:16]: This isn't needed anymore. There is a capitalizePipe. This was overlooked earlier should be fixed now.

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.html [hunk 10]
```diff
@@ -259,11 +131,19 @@
...
+
 <ng-container *ngIf="activeRecord.vehicleType === 'psv'">
   <vtm-vehicle-summary-psv [activeRecord]="activeRecord"></vtm-vehicle-summary-psv>
 </ng-container>
 
-<dl class="govuk-summary-list">
+<ng-container *ngIf="activeRecord.vehicleType === 'trl'">
+  <vtm-vehicle-summary-trl [activeRecord]="activeRecord"></vtm-vehicle-summary-trl>
+</ng-container>
+
```
Comment (alleypa) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.html:145]: we should also have tests to cover these child components from the parents perspective. I can guide here

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.ts (hunk 9 above)
Comment (rspasici) [reply src/app/technical-record/vehicle-summary/vehicle-summary.component.ts:16]: it is for `isStandardVehicleType` on the template

Comment (rspasici) [reply src/app/technical-record/vehicle-summary/vehicle-summary.component.ts:16]: are you talking about ngOnInit ?

File: src/app/technical-record/notes/notes.component.html [hunk 11]
```diff
@@ -1,8 +1,5 @@
 <dl class="govuk-summary-list">
-  <div
-    class="govuk-summary-list__row"
-    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'trl'"
-  >
+  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType !== 'psv'">
```
Comment (m-ujaffer) [inline src/app/technical-record/notes/notes.component.html:2]: I suggest to add unit tests for this

File: src/app/technical-record/notes/notes.component.html [hunk 12]
```diff
@@ -16,7 +13,10 @@
     </dd>
   </div>
 
-  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
+  <div
+    class="govuk-summary-list__row"
```
Comment (m-ujaffer) [inline src/app/technical-record/notes/notes.component.html:17]: unit tests same as above. one test to check for hgv/trl and one for psv should do the job.

File: src/app/technical-record/technical-record.component.ts [hunk 13]
```diff
@@ -17,7 +17,7 @@ import { TechRecord } from './../models/tech-record.model';
 import { MetaData } from '@app/models/meta-data';
 import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
 import { TestResultModel } from '@app/models/test-result.model';
-import { VIEW_STATE } from '@app/app.enums';
+import { VIEW_STATE, VEHICLE_TYPES } from '@app/app.enums';
```
Comment (m-ujaffer) [inline src/app/technical-record/technical-record.component.ts:20]: `VEHICLE_TYPES` is added but not used

File: src/app/technical-record/vehicle-summary/vehicle-summary.component.html [hunk 14]
```diff
@@ -76,147 +72,29 @@
...
-    <dd id="test-speedLimiterMrk" class="govuk-summary-list__value">
-      {{ activeRecord.speedLimiterMrk | DefaultNullOrEmpty }}
+      {{ vehicleClassDescription | DefaultNullOrEmpty | CapitalizeString }}
     </dd>
     <dd class="govuk-summary-list__value"></dd>
   </div>
   <div
     class="govuk-summary-list__row"
-    *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'psv'"
+    *ngIf="techRecHelpers.isStandardVehicle(activeRecord.vehicleType)"
```
Comment (m-ujaffer) [inline src/app/technical-record/vehicle-summary/vehicle-summary.component.html:87]: The old code didn't have trailer. Please revert.

**Comment by m-ujaffer** [comment]
Approved.

**Pull request merged**